crates/autopilot/src/driver_model.rs

```diff
@@ -56,20 +56,32 @@ pub mod solve {
         primitive_types::{H160, U256},
         serde::{Deserialize, Serialize},
         serde_with::{serde_as, DisplayFromStr},
-        std::collections::BTreeMap,
     };
 
     #[serde_as]
     #[derive(Clone, Debug, Default, Serialize)]
     #[serde(rename_all = "camelCase")]
     pub struct Request {
         pub id: i64,
+        pub tokens: Vec<Token>,
         pub orders: Vec<Order>,
-        #[serde_as(as = "BTreeMap<_, DisplayFromStr>")]
-        pub prices: BTreeMap<H160, U256>,
         pub deadline: DateTime<Utc>,
     }
 
+    #[serde_as]
+    #[derive(Clone, Debug, Serialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Token {
+        pub decimals: Option<u8>,
+        pub symbol: Option<String>,
+        pub address: H160,
+        #[serde_as(as = "Option<DisplayFromStr>")]
+        pub price: Option<U256>,
```

Comment:
I think `price`, `address` and `trusted` are the only pieces of information the autopilot has to send. All the other data can get fetched by the `driver` if it's actually needed for its solver.
Also it seems strange to make all prices optional in order to fit in the `trusted` flag somehow. Maybe it makes sense to have 1 entry `prices` as we had before and a list of `trusted_tokens`.

Reply:
Yeah, that's technically correct. Not sure why a `Map<address, price>` felt like stronger guarantee to me. 😅

Reply:
> I think `price` [...] are the only pieces of information the autopilot has to send

Prices are only required for orders that are traded. This `vec` can include tokens that are trusted but not traded (and hence have no price).